### PR TITLE
Enhance tgf to support multiple docker build config

### PR DESCRIPTION
**Pull request by jocgir**
- Create named image instead of anonymous image when building custom image
- Add --prune option to cleanup images
- Fix color rendering problem on Windows.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -87,16 +118,19 @@ func (config TGFConfig) String() (result string) {
 	ifNotZero(dockerImage, config.Image)
```
Comment (julienduchesne) [inline config.go]: This method's name is really not descriptive

File: config.go [hunk 2]
```diff
@@ -87,16 +118,19 @@ func (config TGFConfig) String() (result string) {
...
-	ifNotZero(dockerImageBuildFolder, config.ImageBuildFolder)
-	if config.ImageBuild != "" {
-		lines := strings.Split(strings.TrimSpace(config.ImageBuild), "\n")
-		buildScript := lines[0]
-		if len(lines) > 1 {
-			sep := "\n    "
-			buildScript = sep + strings.Join(lines, sep)
-		}
+	for _, ib := range config.ImageBuild {
+		ifNotZero(ib.Folder, ib.Folder)
```
Comment (julienduchesne) [inline config.go]: Maybe something like "docker-image-build.index" should be given instead of `ib.Folder` twice

File: config.go [hunk 3]
```diff
@@ -87,16 +118,19 @@ func (config TGFConfig) String() (result string) {
...
+		ifNotZero(ib.Tag(), ib.Tag())
```
Comment (julienduchesne) [inline config.go]: Same here

File: main.go [hunk 4]
```diff
@@ -157,56 +187,76 @@ func main() {
...
+		default:
+			printErr("%v", err)
+			return false
+		}
+	}
+	return true
+}
+
+func printErr(format string, args ...interface{}) { ErrPrintln(errorString(format, args...)) }
+func warning(format string, args ...interface{})  { ErrPrintln(warningString(format, args...)) }
```
Comment (julienduchesne) [inline main.go]: to be consistent, it could be printWarning

File: docker.go [hunk 5]
```diff
@@ -174,41 +187,177 @@ func runCommands(commands []string) error {
...
 	}
 
-	var dockerFile string
-	if config.ImageBuild != "" {
-		out := Must(ioutil.TempFile(config.ImageBuildFolder, "DockerFile")).(*os.File)
-		Must(fmt.Fprintf(out, "FROM %s \n%s", config.GetImageName(), config.ImageBuild))
-		Must(out.Close())
-		defer os.Remove(out.Name())
-		dockerFile = out.Name()
+	pruneCmd := exec.Command("docker", "system", "prune", "-f")
```
Comment (julienduchesne) [inline docker.go]: You use the docker client in other places. Have you tried using it here?

File: config.go (hunk 1 above)
Comment (jocgir) [reply config.go]: I changed for addIfNotEmpty (but I am not sure that the function is still used). It was mainly used to print the structure for debugging purposes.

File: config.go (hunk 2 above)
Comment (jocgir) [reply config.go]: Wrong code, thank you!

File: config.go (hunk 3 above)
Comment (jocgir) [reply config.go]: Same too

File: docker.go (hunk 5 above)
Comment (jocgir) [reply docker.go]: I considered it, but was too lazy 😜.

File: main.go (hunk 4 above)
Comment (jocgir) [reply main.go]: Done, changed printErr to printError to be consistent too!

**Pull request merged**